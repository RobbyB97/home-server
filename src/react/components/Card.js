import React from 'react'
import {Link} from 'react-router-dom'

const Card = ({image, title, subtext, link}) => (
    <div>
        <div className="card__wrap">
            <a className="card__link" href={link}>
                <div className="card">
                    <img src={image}/>
                    <div className="card__text">
                        <h2>{title}</h2>
                        <p>{subtext}</p>
                    </div>
                </div>
            </a>
        </div>
    </div>
)

export default Card