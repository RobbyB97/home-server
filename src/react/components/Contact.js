import React from 'react';


export class Contact extends React.Component {
    render() {
        return (
            <section id="Contact" className="contact">
                <div className="contact__shape one"></div>
                <div className="contact__shape two"></div>
                Contact component
            </section>
        );
    }
}


export default Contact;