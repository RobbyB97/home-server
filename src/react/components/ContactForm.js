/* Libraries */
import React from 'react'


export class ContactForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            nameValue: ''
        };

        this.handleNameChange = this.handleNameChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleNameChange(event) {
        this.setState({
            ...this.state,
            nameValue: event.target.value
        });
    }

    handleSubmit(event) {
        console.log('hi')
    }

    render() {
        return (
            <div className="ContactForm">
                <br /><br /><br /><br /><br />
                <form id="contactForm">
                    <p>
                        <label>Name</label>
                        <input type="text" 
                            value={this.state.nameValue} 
                            onChange={this.handleNameChange} 
                            required 
                        />
                    </p>
                </form>
            </div>
        );
    }
}


export default ContactForm