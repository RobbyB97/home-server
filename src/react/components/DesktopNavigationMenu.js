import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {setTitle} from '../actions/title'


const DesktopNavigationMenu = ({dispatch}) => (
    <div id="desktop-navigation-menu">
        <ul>
            <li>
                <NavLink to="/react/" onClick={() => {
                    dispatch(setTitle(''))
                }}>Home</NavLink>
            </li>
            <li>
                <NavLink to="/react/style_guide" onClick={() => {
                    dispatch(setTitle('Style Guide'))
                }}>Style Guide</NavLink>
            </li>
        </ul>
    </div>
)

export default connect()(DesktopNavigationMenu)