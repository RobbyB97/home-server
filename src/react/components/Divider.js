import React from 'react';


export class Divider extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div className={`divider ${this.props.section}`}>
                <div className="divider__shape one"></div>
                <div className="divider__shape two"></div>
            </div>
        );
    }
}


export default Divider;