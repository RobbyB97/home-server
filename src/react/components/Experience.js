import React from 'react';


export class Experience extends React.Component {
    render() {
        return (
            <section id="Experience" className="experience">
                <div className="experience__shape one"></div>
                <div className="experience__shape two"></div>
                Experience component
            </section>
        );
    }
}


export default Experience;