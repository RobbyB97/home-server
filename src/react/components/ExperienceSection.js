import React from 'react';


export class ExperienceSection extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            active: false
        };
    }

    toggle = () => {
        this.setState({
            active: !this.state.active
        });
    };

    render() {
        return (
            <article className="experienceSection">
                <header className="experienceSection__header">
                    {this.props.logo &&
                        <img 
                            className="experienceSection__logo"
                            src={this.props.logo}
                        />
                    }

                    <div className="experienceSection__info">
                        <h2 className="experienceSection__company">
                            {this.props.company}
                        </h2>

                        {this.props.jobTitle &&
                            <h4 className="experienceSection__jobTitle">
                                {this.props.jobTitle}
                            </h4>
                        }

                        {this.props.dates &&
                            <h5 className="experienceSection__dates">
                                {this.props.dates[0]} {this.props.dates[1] && `- ${this.props.dates[1]}`}
                            </h5>
                        }
                    </div>
                </header>
                
                <section 
                    className="experienceSection__description"
                    data-active={this.state.active}
                >
                    <p>{this.props.description}</p>
                </section>

                <footer 
                    className="experienceSection__footer"
                    data-active={this.state.active}
                    onClick={this.toggle}
                >
                    <p>{this.state.active ? "⮝" : "⮟"}</p>
                </footer>
            </article>
        );
    }
}


export default ExperienceSection;