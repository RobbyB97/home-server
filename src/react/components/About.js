import React from 'react';

import Divider from './Divider'


export class About extends React.Component {
    render() {
        return (
            <section id="About" className="about">
                <Divider section="about"/>

                <div className="about__content">
                    <div className="about__portrait">
                        <img src="/dist/images/aboutImage.png" />
                    </div>

                    <article className="about__bio">
                        <p>
                            Hi, my name's Robby. <span className="about__bioTitle">I'm a front-end web developer and freelance digital artist.</span>

                            <br /><br />
                            I put my interdisciplinary skillset to good use by building websites that are as beautiful as they are functional.

                            
                        </p>
                    </article>
                </div>
            </section>
        );
    }
}


export default About;