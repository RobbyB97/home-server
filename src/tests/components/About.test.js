import React from 'react'
import {shallow} from 'enzyme'

import {About} from '../../react/components/About'


test('Render About', () => {
    const wrapper = shallow(<About />)
    expect(wrapper).toMatchSnapshot()
})