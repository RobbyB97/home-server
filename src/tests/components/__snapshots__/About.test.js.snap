// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render About 1`] = `
<section
  className="about"
  id="About"
>
  <Divider
    section="about"
  />
  <div
    className="about__content"
  >
    <div
      className="about__portrait"
    >
      <img
        src="/dist/images/aboutImage.png"
      />
    </div>
    <article
      className="about__bio"
    >
      <p>
        Hi, my name's Robby. 
        <span
          className="about__bioTitle"
        >
          I'm a front-end web developer and freelance digital artist.
        </span>
        <br />
        <br />
        I put my interdisciplinary skillset to good use by building websites that are as beautiful as they are functional.
      </p>
    </article>
  </div>
</section>
`;
